test/integration/smoke/test_hostha_kvm.py

```diff
@@ -274,9 +274,12 @@ def test_hostha_enable_ha_when_host_in_maintenance(self):
             Tests Enable HA when host is in Maintenance mode, should be Ineligible
         """
         self.logger.debug("Starting test_hostha_enable_ha_when_host_in_maintenance")
-
+        self.logger.debug("Pausing to wait for VMs to have finished starting")
+        time.sleep(300)
```

Comment:
Can't we execute this "sleep" differently? I mean, wouldn't it be better to check the "startVm" job to see if it has finished properly?

Reply:
ideally yes.  BUT a. that's beyond my ability (this is the 2nd bit of python i've ever written) and b. Across the test infrastructure, there may be many VMs/VRs and there may be many hosts.  In our Trillian environment there are only 2 hosts by default, so there is a fifty/fifty chance that there was a VR on the host that was fenced and then recovered.
In the end, even trying to cover all of the possibilities seemed excessive for the use case.

Reply:
I though CloudStack moves all VMs/VRs out of a host when preparing for maintenance? Let's work this out in other more elegant way, not a fan of setting things on hold for 5mins just like that..

Reply:
I'm happy for you to work that out, I'm just trying to get a set of tests that aren't constantly throwing false positives, which make them useless.

Reply:
@borisstoyanov indeed CloudStack does that. There is something we could use, the host status. It will only go to maintenance mode when everything is migrated out of the host. Also, if there is a problem and something is not migrated, the host will go to `ErrorInMaitenance` mode.